### PR TITLE
Grouping files by SIGs - SIG API Machinery

### PR DESCRIPTION
**Pull request by eduartua**
Ref. #3064 and #3097 

Here I moved all files owned by SIG API Machinery. Also, all links to those files were fixed(in k/community).

/sig contributor-experience
/cc @parispittman @nikhita 

/assign @cblecker @idvoretskyi @deads2k @lavalamp 
/kind cleanup
/area developer-guide

### CODE REVIEW

**Comment by cblecker** [comment]
The folder shouldn't have caps in the name

**Comment by eduartua** [comment]
Caps in folder name removed.

**Comment by eduartua** [comment]
All URLs were updated - caps removed and changed by lowercase.

**Comment by cblecker** [comment]
this looks good, but a rebase is required.

**Comment by eduartua** [comment]
Rebase done.

File: contributors/devel/api-conventions.md [hunk 1]
```diff
@@ -1,3 +1,1372 @@
+<<<<<<< HEAD
```
Comment (eiais) [inline contributors/devel/api-conventions.md]: This should have been resolved.

**Comment by tallclair** [comment]
This merged with a git conflict. Please fix.

File: contributors/devel/controllers.md [hunk 2]
```diff
@@ -1,191 +1,3 @@
...
-	// method will add a backoff to avoid hotlooping on particular items
-	// (they're probably still not going to work right away) and overall
-	// controller protection (everything I've done is broken, this controller
-	// needs to calm down or it can starve other useful work) cases.
-	c.queue.AddRateLimited(key)
-
-	return true
-}
-```
+This file is a placeholder to preserve links.  Please remove by April 24, 2019 or the release of kubernetes 1.13, whichever comes first.
```
Comment (nikhita) [inline contributors/devel/controllers.md]: release-1.13 has already been released :)

I'm curious why all tombstones have the removal date as April 24, 2019? (sorry, I was away last week and couldn't review other PRs which moved the files in the devel guide. Feel free to point me to a comment/link if this was discussed).

I would prefer to choose a date that's near a release so we can remove all tombstones once a release is done. For example: 1.14 gets released on 25 March 2019.

**Pull request merged**